Core 12.1.0

MingcongBai opened the pull request:
Topic Description
-----------------

- autobuild4: update to 4.7.10
- aosc-aaa: update to 12.1.0
- zstd: update to 1.5.7
    - Use Git snapshot per the Styling Manual.
    - Revised hard-coded references to versions in hwcaps/install.
- zlib: update to 2.2.4
- libxcrypt: update to 4.4.38
- isl: add libtool build dep
- linux+api: update to 6.12.15
- gcc: fix and clean up
    - Backport a fix for ICE on AArch64 whilst building AOM-AV1. \[^1\]
    - \(loongson3\) Drop MIPS LL/SC patch which is no longer needed.
    Track patches at AOSC-Tracking/gcc @ aosc/releases/gcc-14.2.0
    \(HEAD: a386facca6cf12b925cb9185c5bb793af4220c2f\).
    \[^1\]: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=115464
    Co-authored-by: Xi Ruoyao <xry111@xry111.site>
- glibc: add an echo for ldconfig
- glibc: undefine _FILE_OFFSET_BITS, _TIME_BITS to fix build on 32-bit architectures
    Autobuild specifies 64-bit file offset and time bits for 32-bit
    architectures, but glibc handles offset tweaking automatically in its
    build system. Specifying these bits manually will cause builds to fail.
    Undefine these variables to fix build.

Package(s) Affected
-------------------

- aosc-aaa: 12.1.0
- autobuild4: 4.7.10
- gcc: 14.2.0-3
- gcc-runtime: 14.2.0-3
- glibc: 1:2.40-2
- isl: 1:0.27-3
- libxcrypt: 4.4.38
- linux+api: 2:6.12.15
- zlib: 1:2.2.4
- zstd: 1.5.7

Security Update?
----------------

No

Build Order
-----------

```
#buildit autobuild4 linux+api glibc libxcrypt isl zlib zstd gcc aosc-aaa
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

MingcongBai committed:
glibc: add an echo for ldconfig

MingcongBai committed:
gcc: fix and clean up

- Backport a fix for ICE on AArch64 whilst building AOM-AV1. [^1]
- (loongson3) Drop MIPS LL/SC patch which is no longer needed.

Track patches at AOSC-Tracking/gcc @ aosc/releases/gcc-14.2.0
(HEAD: a386facca6cf12b925cb9185c5bb793af4220c2f).

[^1]: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=115464

Co-authored-by: Xi Ruoyao <xry111@xry111.site>

MingcongBai committed:
linux+api: update to 6.12.15

MingcongBai committed:
isl: add libtool build dep

MingcongBai committed:
libxcrypt: update to 4.4.38

MingcongBai committed:
zlib: update to 2.2.4

MingcongBai committed:
zstd: update to 1.5.7

- Use Git snapshot per the Styling Manual.
- Revised hard-coded references to versions in hwcaps/install.

MingcongBai committed:
aosc-aaa: update to 12.1.0

MingcongBai committed:
autobuild4: update to 4.7.10

xry111 commented on core-devel/gcc/01-runtime/patches/0003-aarch64-Add-another-use-of-force_subreg-PR115464.patch:
This won't build due to missing https://github.com/AOSC-Tracking/gcc/commit/4662fb90aebe688c1885ee86b5b40d1e1c072455.

MingcongBai replied:
Got it, oops.

MingcongBai committed:
glibc: undefine _FILE_OFFSET_BITS, _TIME_BITS to fix build on 32-bit architectures

Autobuild specifies 64-bit file offset and time bits for 32-bit
architectures, but glibc handles offset tweaking automatically in its
build system. Specifying these bits manually will cause builds to fail.

Undefine these variables to fix build.

xry111 commented on core-devel/gcc/01-runtime/patches/0002-MIPS-Include-missing-mips16.S-in-libgcc-lib1funcs.S.patch:
Ok, needed to prevent things from "requiring exec stack."

xry111 commented on core-devel/gcc/01-runtime/patches/0003-aarch64-Fix-invalid-nested-subregs-PR115464.patch:
Ok, prerequisite for the ICE fix.

xry111 commented on core-devel/gcc/01-runtime/patches/0004-aarch64-Add-another-use-of-force_subreg-PR115464.patch:
Ok, an ICE fix.

xry111 commented on core-devel/gcc/01-runtime/patches/0001-libiberty-also-generate-a-PIC-version-of-libiberty.a.patch:
I'm not sure what this is for...

xry111 replied:
So if `--enable-install-libiberty` is still required, this is required.  I don't like `--enable-install-libiberty` but I have no time to investigate if it can be removed now.

xry111 committed:
gcc: [NFC] Remove incorrect comment

The GNU hash style enabling seems (unintentionally?) removed at commit
764546ea331a1c1baecc53c5caafc4c6741c2f8a.  Now we are going to enable it
at the Binutils side instead (see #9531) so just remove the comment
here.

xry111 reviewed (approved):
Tested on loongson3.

MingcongBai commented:
Appending LGTM from AMD64, AArch64, and LoongArch.
